challenge3.py

```diff
@@ -0,0 +1,51 @@
+"""
+   Advent of Code 2018 Day 3
+"""
+from itertools import product, chain, groupby
+import re
+
+from common.input_file import get_transformed_input
+
+def get_claim(claim):
+    """
+        Transform a claim to an ID and a list of points
+    """
+    claim_id, left, top, width, height = re.split(r"@|,|:|x", claim)
+    left, top, width, height = int(left), int(top), int(width), int(height)
+    points = product(range(left, left + width), range(top, top + height))
+    return (claim_id, list(points))
```

Comment:
I like the implementation of just storing the list of points associated with a claim rather than just the metadata. I was worried about storage initially, but I think the clarity that this causes elsewhere is a much higher benefit than the negative of the small memory hit